SimTKcommon/src/Pathname.cpp

```diff
@@ -110,94 +110,117 @@ static void removeDriveInPlace(string& inout, string& drive) {
-// That is then removed for further processing.
-// Absolute paths are designated like this:
-//      Leading "/" means root relative (on the drive).
-//      Leading "./" means current working directory relative (on the drive).
-//      Leading "../" is interpreted as "./..".
-//      Leading "@/" means relative to executable location.
-// Above leading characters are removed and replaced with the
-// full path name they represent, then the entire path is divided
-// into components. If a component is "." or "" (empty) it is
-// removed. If a component is ".." it and the previous component
-// if any are removed. (".." as a first component will report as
-// ill formed.)
-// If there is something ill-formed about the file name we'll return
-// false.
-void Pathname::deconstructPathname( const string&   name,
-                                    bool&           isAbsolutePath,
-                                    string&         directory,
-                                    string&         fileName,
-                                    string&         extension)
+void Pathname::deconstructPathnameUsingSpecifiedWorkingDirectory(const std::string& swd,
+                                            const std::string& path,
+                                            std::string& directory,
+                                            std::string& fileName,
+                                            std::string& extension,
+                                            bool& dontApplySearchPath)
 {
-    isAbsolutePath = false;
     directory.erase(); fileName.erase(); extension.erase();
+    string pathdrive, swddrive, finaldrive;
+    dontApplySearchPath = false;
 
     // Remove all the white space and make all the slashes be forward ones.
     // (For Windows they'll be changed to backslashes later.)
-    String processed = String::trimWhiteSpace(name).replaceAllChar('\\', '/');
-    if (processed.empty())
-        return; // name consisted only of white space
-
-    string drive;
-    removeDriveInPlace(processed, drive);
-
-    // Now the drive if any has been removed and we're looking at
-    // the beginning of the path name. 
+    String pathcleaned = String::trimWhiteSpace(path).replaceAllChar('\\', '/');
+    if (pathcleaned.empty())
+        return; // path consisted only of white space
+    removeDriveInPlace(pathcleaned, pathdrive);
+
+    // Check for a drive, then map swd to an absolute path name if no drive given
+    String swdcleaned = String::trimWhiteSpace(swd).replaceAllChar('\\', '/');
+    if (!swdcleaned.empty() && swdcleaned[swdcleaned.size() - 1] != '/')
+        swdcleaned += '/';
+    removeDriveInPlace(swdcleaned, swddrive);
+
+    String processed;
+    if (pathcleaned.substr(0, 1) == "/" || pathcleaned.substr(0, 2) == "@/") {
+        processed = pathcleaned;
+        if (!pathdrive.empty()) swddrive = "";
+    }
+    else if (pathcleaned.substr(0, 2) == "./") {
+        if (!swd.empty()) {
```

Comment:
`swd` was not cleaned, so it will not test empty even if it is just a bunch of blanks. So I don't think this is the right test. That also suggests that the test suite should include some tests where `swd` is non-empty but just whitespace.